bitsandbytes/nn/modules.py

```diff
@@ -186,9 +193,15 @@ def to(self: T, tensor: Tensor, non_blocking: bool = ...) -> T:
 
     def to(self, *args, **kwargs):
         device, dtype, non_blocking, convert_to_format = torch._C._nn._parse_to(*args, **kwargs)
-
-        if (device is not None and device.type == "cuda" and self.data.device.type == "cpu"):
-            return self.cuda(device)
+    
+        if device is not None and device.type == "cpu":
+            return self.cpu(device)
+
+        if (device is not None and device.type != "cpu" and self.data.device.type == "cpu"):
+            if device.type == "cuda":
+                return self.cuda(device)
+            elif device.type == "xpu":
+                return self.xpu(device)
```

Comment:
made changes to use common quant_to_device for dispatch, since we will support each device's quantization, the specific self.cuda or self.xpu is not needed.